lib/config.go

```diff
@@ -9,29 +9,18 @@ import (
 )
 
 // ConfigMethods encapsulates changes to a qri configuration
-type ConfigMethods interface {
-	// CoreRequestsName specifies participation in the methods interface
-	// TODO (b5): update func name when change happens
-	CoreRequestsName() string
-	// GetConfig returns the Config, or one of the specified fields of the Config,
-	// as a slice of bytes the bytes can be formatted as json, concise json, or yaml
-	GetConfig(p *GetConfigParams, res *[]byte) (err error)
-	// SetConfig validates, updates and saves the config
-	SetConfig(update *config.Config, set *bool) (err error)
+type ConfigMethods struct {
+	inst *Instance
 }
 
+
 // NewConfigMethods creates a configuration handle from an instance
 func NewConfigMethods(inst *Instance) ConfigMethods {
-	return configMethods{inst: inst}
-}
-
-// configMethods is the private implementation of ConfigMethods
-type configMethods struct {
-	inst *Instance
+	return ConfigMethods{inst: inst}
```

Comment:
One new-ish convention is now exposed: the returned `ConfigMethods` is a value, not a pointer, because ConfigMethods can't change any of it's internal values.

One concern I have with exposing this: we won't be able to get away with returning values for all "Methods" structs, because some structs need additional state to do their work (`RemoteRequests` is a good example). 

This begs a choice: 
Should we favour consistency and  return `*ConfigMethods`, even though all Methods are not defined on a pointer? Or should we favour accuracy, and return the struct value?

Reply:
After doing a quick audit, only `ConfigMethods` and `ProfileMethods` don't return pointers. I think we should just be consistent, as long as we aren't losing anything.

Reply:
dope. let's do it!